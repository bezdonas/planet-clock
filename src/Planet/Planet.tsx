import React from 'react';
import styled from 'styled-components';
import Clock from '../Clock/Clock';
import { Time } from 'src/types/Clock';

const StyledPlanetWrapper = styled.div`
  color: white;
  margin: 15px;
  align-self: center;
  justify-self: center;
  h4 {
    text-align: center;
  }
`;

export type PlanetProps = {
  title: string;
  description?: string;
  secondDuration: number;
} & Time;

export default class Planet extends React.PureComponent<PlanetProps> {
  public render() {
    const { title, secondDuration, hour, minute, second } = this.props;
    return (
      <StyledPlanetWrapper>
        <Clock
          secondDuration={secondDuration}
          hour={hour}
          minute={minute}
          second={second}
        />
        <h4>{title}</h4>
      </StyledPlanetWrapper>
    );
  }
}
