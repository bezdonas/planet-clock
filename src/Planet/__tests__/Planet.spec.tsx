import React from 'react';
import { shallow } from 'enzyme';
import Planet from '../Planet';

jest.mock('moment', () => () => ({
  minute: () => 20,
  hour: () => 42,
  second: () => 12,
}));

it('matches snapshot', () => {
  const props = {
    title: 'Earth',
    secondDuration: 1000,
  };

  expect(shallow(<Planet {...props} />)).toMatchSnapshot();
});
