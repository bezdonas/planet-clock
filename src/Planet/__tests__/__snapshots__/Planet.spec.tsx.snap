// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Planet
    secondDuration={1000}
    title="Earth"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Clock
          hour={42}
          minute={20}
          second={12}
          secondDuration={1000}
/>,
        <h4>
          Earth
</h4>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "hour": 42,
          "minute": 20,
          "second": 12,
          "secondDuration": 1000,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Earth",
        },
        "ref": null,
        "rendered": "Earth",
        "type": "h4",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": true,
        "rules": Array [
          "
  color: white;
  margin: 15px;
  align-self: center;
  justify-self: center;
  h4 {
    text-align: center;
  }
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Clock
            hour={42}
            minute={20}
            second={12}
            secondDuration={1000}
/>,
          <h4>
            Earth
</h4>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "hour": 42,
            "minute": 20,
            "second": 12,
            "secondDuration": 1000,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Earth",
          },
          "ref": null,
          "rendered": "Earth",
          "type": "h4",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": true,
          "rules": Array [
            "
  color: white;
  margin: 15px;
  align-self: center;
  justify-self: center;
  h4 {
    text-align: center;
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
